components/modal/index.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Component, createPortal } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import ModalFrame from './frame';
+import ModalHeader from './header';
+import * as ariaHelper from './aria-helper';
+import './style.scss';
+
+// Used to count the number of open modals.
+let parentElement,
+	openModalCount = 0;
+
+class Modal extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.prepareDOM();
+	}
+
+	/**
+	 * Prepares the DOM for the modals to be rendered.
+	 *
+	 * Every modal is mounted in a separate div appended to a parent div
+	 * that is appended to the document body.
+	 *
+	 * The parent div will be created if it does not yet exist, and the
+	 * separate div for this specific modal will be appended to that.
+	 */
+	prepareDOM() {
+		if ( ! parentElement ) {
+			parentElement = document.createElement( 'div' );
+			document.body.appendChild( parentElement );
+		}
+		this.node = document.createElement( 'div' );
+		parentElement.appendChild( this.node );
+	}
+
+	/**
+	 * Removes the specific mounting point for this modal from the DOM.
+	 */
+	cleanDOM() {
+		parentElement.removeChild( this.node );
+	}
+
+	/**
+	 * Opens or closes the modal based on whether the isOpen prop changed.
+	 *
+	 * @param {Object} prevProps The previous props.
+	 */
+	componentDidUpdate( prevProps ) {
+		const openStateChanged = this.props.isOpen !== prevProps.isOpen;
```

Comment:
This will become - `componentDidMount` - which contains only `this.openModal();` call.